src/entropic/sinkhorn_epsscaling.jl

```diff
@@ -0,0 +1,171 @@
+struct SinkhornEpsilonScaling{A<:Sinkhorn,T<:Real} <: Sinkhorn
+    alg::A
+    factor::T
+    steps::Int
+end
+
+function Base.show(io::IO, alg::SinkhornEpsilonScaling)
+    return print(
+        io, alg.alg, " with ε-scaling (steps: ", alg.steps, ", factor: ", alg.factor, ")"
+    )
+end
+
+"""
+    SinkhornEpsilonScaling(algorithm::Sinkhorn; factor=1//2, steps=5)
+
+Construct an ε-scaling Sinkhorn algorithm for solving an entropically regularized optimal
+transport problem.
+
+The function uses the specified Sinkhorn `algorithm` with `steps` ε-scaling steps
+with scaling factor `factor`. It sequentially solves the entropically regularized
+optimal transport with regularization parameters
+```math
+\\varepsilon_i := \\varepsilon \\lambda^{i-k}, \\qquad (i = 1,\\ldots,k),
+```
+where ``\\lambda`` is the scaling factor and ``k`` the number of scaling steps.
+"""
+function SinkhornEpsilonScaling(alg::Sinkhorn; factor::Real=1//2, steps::Int=5)
```

Comment:
It does not promote which is relevant e.g. if you want to keep `Float32` (otherwise they are promoted to `Float64` which leads to terrible performance with GPUs).